packages/rs-drive-abci/src/execution/engine/run_block_proposal/v0/mod.rs

```diff
@@ -139,6 +140,12 @@ where
                 .expect("current epoch index should be in range"),
         );
 
+        // Patch platform version to fix chain halt bugs
+        patch_platform_version(&block_info, platform_version, &mut block_platform_state)?;
+
+        // Perform state migration to fix bugs or support new features
+        self.migrate_state(&block_info, &mut block_platform_state)?;
```

Comment:
This should be one call only. We should call it, check_for_chain_halt_hot_fix.

Reply:
And it should do everything inside.

Reply:
Migrations aren't only for chain halts. We might want to migrate data for features as well.